File: clojure/eu001.clj
(defn div3or5? 
  "tests if x is divisble by 3 or 5"
  [x]
  (or (= (rem x 3) 0) 
      (= (rem x 5) 0)))

(defn sum-div-3or5 
  "Computes sum of all integers below n that are 
  divisble by 3 or 5"
  [n]
  (reduce + (filter div3or5? (range n))))

(def run-test '(= (sum-div-3or5 10) 23))
(def run-problem '(sum-div-3or5 1000))
;(sum-div-3or5 1000) ;=> 233168

File: zig/eu001.zig
const std = @import("std");
const expect = @import("std").testing.expect;

fn natMult3or5(max: i32) i32 {
    var result: i32 = 0;
    var i: i32 = 0;
    while (i < max): (i+=1) {
        if (@mod(i,3) == 0 or @mod(i,5) == 0) {
            result += i;
        }
    }
    return result;
}

test "sum of natural numbers below 10 that are multiples of 3 or 5" {
    expect(natMult3or5(10) == 23);
}

pub fn main() void {
    std.debug.print("{}\n",.{natMult3or5(@as(i32,1000))});
}
